src/pages/settings/Wallet/ReportFraudPage.js

```diff
@@ -0,0 +1,110 @@
+import React, {useEffect} from 'react';
+import _ from 'underscore';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ROUTES from '../../../ROUTES';
+import HeaderWithBackButton from '../../../components/HeaderWithBackButton';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+import Navigation from '../../../libs/Navigation/Navigation';
+import styles from '../../../styles/styles';
+import Text from '../../../components/Text';
+import useLocalize from '../../../hooks/useLocalize';
+import * as Card from '../../../libs/actions/Card';
+import assignedCardPropTypes from './assignedCardPropTypes';
+import * as CardUtils from '../../../libs/CardUtils';
+import ONYXKEYS from '../../../ONYXKEYS';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import Form from '../../../components/Form';
+import usePrevious from '../../../hooks/usePrevious';
+import * as FormActions from '../../../libs/actions/FormActions';
+
+const propTypes = {
+    /* Onyx Props */
+    formData: PropTypes.shape({
+        isLoading: PropTypes.bool,
+    }),
+    cardList: PropTypes.objectOf(assignedCardPropTypes),
+    /** The parameters needed to authenticate with a short-lived token are in the URL */
+    route: PropTypes.shape({
+        /** Each parameter passed via the URL */
+        params: PropTypes.shape({
+            /** Domain string */
+            domain: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+const defaultProps = {
+    cardList: {},
+    formData: {},
+};
+
+function ReportFraudPage({
+    route: {
+        params: {domain},
+    },
+    cardList,
+    formData,
+}) {
+    const {translate} = useLocalize();
+
+    const domainCards = CardUtils.getDomainCards(cardList)[domain];
+    const virtualCard = _.find(domainCards, (card) => card.isVirtual) || {};
+
+    const prevIsLoading = usePrevious(formData.isLoading);
+
+    useEffect(() => {
+        if (prevIsLoading && formData.isLoading === false && _.isEmpty(virtualCard.errors)) {
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
+        }
+
+        if (formData.isLoading !== false && _.isEmpty(virtualCard.errors)) {
+            return;
+        }
+
+        FormActions.setErrors(ONYXKEYS.FORMS.REPORT_FRAUD_FORM, virtualCard.errors);
```

Comment:
I don't think this is a pattern that we follow. The backend errors are not form errors

Reply:
+1 the errors should either come from the backend (in that case they are already there on the `virtualCard` object) or the form itself should set them. In this case though, we have no form fields so there is no validation that could fail.

Reply:
Basically the options are:

1. Use the FormAlertWithSubmitButton directly (I think this might be the way I would go here) and then pass the errors that are on the card object.
2. Use the Form - but don't manually set the errors. Just trust that the backend will set them if anything goes wrong.

Reply:
All right, so let's go with number 1! What exactly do you mean by "pass the errors that are on the card object."?

Reply:
I think I get it now, let me know if that's what you meant 😄 CC: @marcaaron